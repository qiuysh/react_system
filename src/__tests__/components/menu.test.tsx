/** @format */
import * as React from "react";
// import { shallow, render } from "enzyme";
// import "jest-canvas-mock";
import * as renderer from "react-test-renderer";
import SiderMenu from "@components/siderMenu";

describe("菜单测试", () => {
  it("siderMenu -- 快照", () => {
    const props = {
      collapsed: false,
      history: {
        location: {
          pathname: "/",
        },
      },
      menuList: [
        {
          id: 1,
          pid: 0,
          name: "仪表盘",
          type: 1,
          icon: "icondashboard",
          url: "/dashboard",
          code: "dashboard",
          desc: null,
          create_time: "2021-03-10T10:47:52.000Z",
          modified_time: null,
        },
      ],
    };
    const wrapper = renderer
      .create(<SiderMenu {...props} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
