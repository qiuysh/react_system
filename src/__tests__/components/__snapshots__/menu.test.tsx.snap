// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`菜单测试 siderMenu -- 快照 1`] = `
<aside
  className="ant-layout-sider ant-layout-sider-dark yux-sider"
  style={
    Object {
      "flex": "0 0 200px",
      "maxWidth": "200px",
      "minWidth": "200px",
      "width": "200px",
    }
  }
>
  <div
    className="ant-layout-sider-children"
  >
    <div
      className="logo"
    >
      Wolin 3C
    </div>
    <ul
      className="ant-menu ant-menu-dark ant-menu-root ant-menu-inline"
      onMouseEnter={[Function]}
      onTransitionEnd={[Function]}
      role="menu"
      style={Object {}}
    >
      <li
        className="ant-menu-item ant-menu-item-only-child ant-menu-item-selected"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 24,
          }
        }
      >
        <span>
          仪表盘
        </span>
      </li>
    </ul>
  </div>
</aside>
`;
